### ISSUE TITLE
Youtube videos & html5 video, error on exiting full screen video playback

### ISSUE THREAD

**Issue by anjimi** [Simbul/baker#611]
In our magazine app we have some videos, a couple are Youtube iframe embeds and a couple which are included files embedded with html5 video.

We switched to the new Baker 4.0 Newsstand version recently, and there is a difference in how the Youtube videos are played on the iPad.

In Baker 3.2 the Youtube videos play in the same (iOS native?) player as the included video files.

In Baker 4.0 the Youtube videos play in a Youtube player, and there is no full-screen option.

The behaviour from 3.2 is preferred, where all videos play in the same player.

Does anyone know why this would have changed, and is there anything I can do to get the Youtube videos to play in the same player as included video files?

**Comment by roninboy** [comment]
If you remove the User Agent info in the App Delegate File - you will see the Youtube player reverting to the older version - which allows full screen etc. Dont know if you require the User Agent info for any other functionality.

**Comment by anjimi** [comment]
Thank you @roninboy that worked for the youtube video player for me. 

What other functionality could the User Agent be needed for?

There is one issue with the video player in iOS 6, when exiting full screen video, instead of continuing playback of embedded video in the page, the page goes back to the top and video playback is stopped and returned to the start of the video, as if the page has completely reloaded.

This doesn't happen in iOS 5.1, it behaves as expected. Any ideas on that?

**Comment by roninboy** [comment]
I think the User Agent info was added to Baker originally to support Webkit-based animation (exporting from Hype) - but I suppose there's any number of ways to use it to customise the look and use of an app on a particular device. If you havent specified its use yourself within your app then dont worry about it - I dont think its required elsewhere in Baker.

I dont seem to be able to replicate the error you're describing with the full screen video. Are you using the old embed code for videos or the video tag?

**Comment by anjimi** [comment]
Ok, thanks for the info on the User Agent.

The error on exiting full screen video occurs in our app in Baker 4 running in iOS 6 only, and with Youtube linked video or html5 video with video file included.

I don't get the error with Baker 4 in iOS 5.1 or Baker 3.2 in iOS 6.

It's particularly noticeable with our app as our pages have vertical pagination, when you exit full screen video, playback should continue with the video in the page, but it stops and the page reloads to the top.

**Comment by roninboy** [comment]
Still cannot replicate the problem on full screen exit - I've got video working ok on ios6 on all orientations and vertical pagination using the following (standard Video tag for bundled video and old flash embed for youtube - NOT IFRAMES - ios6 - problems with IFrames). Sometimes a clean rebuild - killing all build files and cleaning cache can solve erratic issues. You could also try a different render mode instead of three cards if you're having pagination issues (though I doubt its the problem). If you have a lot of stuff going on the one page - images, sliders, animation, video etc you may need to give it a while to load before jumping in and out of pages or elements.

<pre><code>&lt;video src="video.mp4" controls&gt;

&lt;object width="560" height="315"&gt;&lt;param name="movie" value="http://www.youtube.com/v/zvCBSSwgtg4?hl=en_US&amp;version=3&amp;rel=0"&gt;&lt;/param&gt;&lt;param name="allowFullScreen" value="true"&gt;&lt;/param&gt;&lt;param name="allowscriptaccess" value="always"&gt;&lt;/param&gt;&lt;embed src="http://www.youtube.com/v/zvCBSSwgtg4?hl=en_US&amp;version=3&amp;rel=0" type="application/x-shockwave-flash" width="560" height="315" allowscriptaccess="always" allowfullscreen="true"&gt;&lt;/embed&gt;&lt;/object&gt;
</pre></div>

**Comment by anjimi** [comment]
After much testing I am still getting this problem on exiting full screen video playback.

I've tried the youtube videos with both the new iframe method and the old object method, but as the same problem occurs with videos included with html5 video it's appears to be unrelated to the video source or embed method.

I have tried both screenshot and three-cards rendering modes, with vertical pagination on and off.

I am using latest Baker 4 build, downloaded yesterday. Problem only occurs when testing in iOS 6 iPhone & iPad, actual devices & simulators. No problem in iOS 5.1 iPhone & iPad, actual devices & simulators.

I also have a test version with the exact same book pages in Baker 3.2 which works fine in both iOS 6 & 5.1.

I've tried a test page in my book with nothing but video on page - same problem.
Also tried adding a video to the Baker A Study in Scarlet - same problem.

Steps to reproduce behaviour:
1. play a video, either html5 with included video file or an embedded youtube video
2. tap the double-arrow icon bottom right of video to launch in full screen
3. tap the double-arrow icon to go back to normal view

Video playback should continue, but is stopped, and the whole page reloads. If you tap the 'Done' button in full screen mode, which should go back to normal view but stop playback, the same behaviour occurs with the page reloading.

I have also found that the positioning of the index on double-tap is messed up after exiting full screen video. If you repeat the steps above the same thing happens with the video and page reload, but the index position is then fixed.

Also, very weird, when in screenshot rendering mode, it seems to mess with the screenshot for the page, if you swipe to another page and back again. I'm using my own pre-rendered screenshots and instead of showing the screenshot for the video page it shows the page reloading.

Note: I have deleted the following from AppDelegate.m which makes youtube videos play in the native video player same as html5 video.

> - (void)initialize {
>   // Set user agent (the only problem is that we can't modify the User-Agent later in the program)
>   // We use a more browser-like User-Agent in order to allow browser detection scripts to run (like Tumult Hype).
>   NSDictionary *userAgent = [[NSDictionary alloc] initWithObjectsAndKeys:@"Mozilla/5.0 (compatible; BakerFramework) AppleWebKit/533.00+ (KHTML, like Gecko) Mobile", @"UserAgent", nil];
>   [[NSUserDefaults standardUserDefaults] registerDefaults:userAgent];
>   [userAgent release];
>   }

**Comment by aspek** [comment]
Waiting for a long time about this issue.
But I'm not good at English.

**Comment by roninboy** [comment]
Sorry Anjimi

Finally replicated this issue on a brand new 4.0 install - I have been working with a version which has been altered somewhat for a standalone single book (based on 3.22) - which doesnt seem to have a problem in ios6. 

I've had a look at the logs and had a trawl through the code but I'm still none the wiser.

Apple has made some undocumented changes to the MPMoviePlayerController class in ios6, when you tap on full screen button, the viewWillDisappear and viewDidDisappear gets called for both the controller launching MPMoviePLayerController and its ParentController. I'm not sure if this is relevant for videos launched via UIWebView using the MPAVController and UIMoviePlayerController (as would be the case for say Youtube vids). 

I see the book status is saved inside the viewWillDisappear - and I think updateBookLayout is called before the page reloads. There doesnt seem to be any instance of viewDidDisappear in the 4.0 BakerViewController - but theres definitely no mention of either in Baker 3.22.

I'll keep looking - but I'm stumped for now.

The line below - Is at the point of tapping the fullscreen button of the media player:

2013-01-07 14:29:13.827 Baker[1011:1a003] saved status

Any core help on this would be great.

Full Log:

013-01-07 14:27:55.861 Baker[1011:1a003] [MPAVController] Autoplay: Enabling autoplay
2013-01-07 14:27:55.865 Baker[1011:1a003] [MPCloudAssetDownloadController] Prioritization requested for media item ID: 0
2013-01-07 14:27:57.969 Baker[1011:1a003] [MPAVController] Autoplay: Skipping autoplay, disabled (for current item: 0, on player: 1)
2013-01-07 14:27:57.982 Baker[1011:1a003] [MPAVController] Autoplay: Enabling autoplay
2013-01-07 14:27:57.982 Baker[1011:1a003] setting movie path: http://r1---sn-aigeznl6.c.youtube.com/videoplayback?ipbits=8&sver=3&id=986f17197301d205&expire=1357591313&itag=22&ratebypass=yes&key=yt1&cp=U0hUTFVUT19GTUNONF9NTlNEOjN6aUNvYWI4ZGZa&fexp=900223%2C907065%2C911115%2C916625%2C920704%2C912806%2C928001%2C922403%2C922405%2C929901%2C913605%2C929104%2C913546%2C913556%2C908496%2C920201%2C913302%2C919009%2C911116%2C926403%2C901451%2C902556&ip=86.169.203.223&mv=m&sparams=cp%2Cid%2Cip%2Cipbits%2Citag%2Cratebypass%2Csource%2Cupn%2Cexpire&source=youtube&upn=7JyTIH5RuxU&ms=au&mt=1357568835&signature=9CA2B738074BAA2E67ECB374A179324BD9F78F5F.6E0AA4A5214D6F57929D3A8CF9F6E2AED381B285
2013-01-07 14:27:57.983 Baker[1011:1a003] [MPAVController] Autoplay: Disabling autoplay for pause
2013-01-07 14:27:57.983 Baker[1011:1a003] [MPAVController] Autoplay: Disabling autoplay
2013-01-07 14:27:58.070 Baker[1011:1a003] [MPAVController] Autoplay: _streamLikelyToKeepUp: 0 -> 1
2013-01-07 14:27:58.071 Baker[1011:1a003] [MPAVController] Autoplay: Skipping autoplay, disabled (for current item: 1, on player: 1)
2013-01-07 14:27:58.073 Baker[1011:1a003] [MPAVController] Autoplay: Enabling autoplay
2013-01-07 14:28:25.402 Baker[1011:1a003] ‚Ä¢ User tap at [296.000000, 524.000000]
2013-01-07 14:28:28.844 Baker[1011:1a003] ‚Ä¢ User tap at [72.000000, 809.000000]
2013-01-07 14:28:28.845 Baker[1011:1a003] [MPAVController] Autoplay: Disabling autoplay for pause
2013-01-07 14:28:28.845 Baker[1011:1a003] [MPAVController] Autoplay: Disabling autoplay
2013-01-07 14:28:28.967 Baker[1011:1a003] [MPAVController] Autoplay: _streamLikelyToKeepUp: 1 -> 1
2013-01-07 14:28:45.156 Baker[1011:1a003] ‚Ä¢ User tap at [67.000000, 806.000000]
2013-01-07 14:28:45.510 Baker[1011:1a003] [MPAVController] Autoplay: Enabling autoplay
2013-01-07 14:29:11.874 Baker[1011:1a003] ‚Ä¢ User tap at [526.000000, 759.000000]
2013-01-07 14:29:13.404 Baker[1011:1a003] ‚Ä¢ User tap at [529.000000, 809.000000]
2013-01-07 14:29:13.827 Baker[1011:1a003] saved status
2013-01-07 14:29:13.834 Baker[1011:1a003] [MPAVController] Autoplay: Enabling autoplay
2013-01-07 14:30:52.424 Baker[1011:1a003] ‚Ä¢ Set size for orientation: portrait
2013-01-07 14:30:52.424 Baker[1011:1a003] ‚Ä¢ Set tappable area size
2013-01-07 14:30:52.424 Baker[1011:1a003]     Prevent page from changing until layout is updated
2013-01-07 14:30:52.424 Baker[1011:1a003] ‚Ä¢ Show page details for the book pages
2013-01-07 14:30:52.426 Baker[1011:1a003]     Unlock page changing
2013-01-07 14:30:52.426 Baker[1011:1a003] ‚Ä¢ Init page details for the book pages
2013-01-07 14:30:52.481 Baker[1011:1a003] Set IndexView size to 768x1024, with pageY set to 0
2013-01-07 14:30:52.482 Baker[1011:1a003] Path to index view is /Users/kevin/Library/Application Support/iPhone Simulator/6.0/Applications/DBC3E4F3-FC8B-4621-B479-BC8C96427928/Baker.app/books/City of Music/index.html
2013-01-07 14:30:52.483 Baker[1011:1a003] Path to index view is /Users/kevin/Library/Application Support/iPhone Simulator/6.0/Applications/DBC3E4F3-FC8B-4621-B479-BC8C96427928/Baker.app/books/City of Music/index.html
2013-01-07 14:30:52.483 Baker[1011:1a003] ‚Ä¢ Add page to the loding queue
2013-01-07 14:30:52.483 Baker[1011:1a003] ‚Ä¢ Add page to the loding queue
2013-01-07 14:30:52.483 Baker[1011:1a003] ‚Ä¢ Add page to the loding queue
2013-01-07 14:30:52.483 Baker[1011:1a003] ‚Ä¢ Handle loading of slot 0 with page 16
2013-01-07 14:30:52.483 Baker[1011:1a003] ‚Ä¢ Setup new page for loading
2013-01-07 14:30:52.485 Baker[1011:1a003] ‚Ä¢ Setup webView
2013-01-07 14:30:52.485 Baker[1011:1a003] ‚Ä¢ Loading: book/Page07.html
2013-01-07 14:30:52.776 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:52.776 Baker[1011:1a003]     Page is current page and current page IS delaying loading --> load page
2013-01-07 14:30:52.776 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:52.796 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:52.797 Baker[1011:1a003]     Page is current page and current page IS NOT delaying loading --> handle clicked link: http://www.youtube.com/embed/7ACLLTMgsnw?version=3&hl=en_US&rel=0&showinfo=0
2013-01-07 14:30:52.797 Baker[1011:1a003]     Opening absolute URL: http://www.youtube.com/embed/7ACLLTMgsnw?version=3&hl=en_US&rel=0&showinfo=0
2013-01-07 14:30:52.797 Baker[1011:1a003]     Link doesn't contain param "referrer=Safari" --> open link in page
2013-01-07 14:30:52.797 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:52.799 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:52.799 Baker[1011:1a003]     Page is current page and current page IS NOT delaying loading --> handle clicked link: http://www.youtube.com/embed/CZOhhbewFZ0?version=3&hl=en_US&rel=0&showinfo=0
2013-01-07 14:30:52.799 Baker[1011:1a003]     Opening absolute URL: http://www.youtube.com/embed/CZOhhbewFZ0?version=3&hl=en_US&rel=0&showinfo=0
2013-01-07 14:30:52.799 Baker[1011:1a003]     Link doesn't contain param "referrer=Safari" --> open link in page
2013-01-07 14:30:52.800 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:52.802 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:52.803 Baker[1011:1a003]     Page is current page and current page IS NOT delaying loading --> handle clicked link: http://www.youtube.com/embed/mG8XGXMB0gU?version=3&hl=en_US&rel=0&showinfo=0
2013-01-07 14:30:52.803 Baker[1011:1a003]     Opening absolute URL: http://www.youtube.com/embed/mG8XGXMB0gU?version=3&hl=en_US&rel=0&showinfo=0
2013-01-07 14:30:52.803 Baker[1011:1a003]     Link doesn't contain param "referrer=Safari" --> open link in page
2013-01-07 14:30:52.803 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:53.024 Baker[1011:1a003] Set size for IndexView to 768x205 (constrained from 36490x205)
2013-01-07 14:30:53.058 Baker[1011:1a003] ‚Ä¢ Page did finish load
2013-01-07 14:30:53.058 Baker[1011:1a003] ‚Ä¢ Setting current page height from 1024 to 10118.000000
2013-01-07 14:30:53.058 Baker[1011:1a003] ‚Ä¢ webView hidden: 0 animating: 1
2013-01-07 14:30:53.059 Baker[1011:1a003] ‚Ä¢ Handle loading of slot 1 with page 17
2013-01-07 14:30:53.059 Baker[1011:1a003] ‚Ä¢ Setup new page for loading
2013-01-07 14:30:53.060 Baker[1011:1a003] ‚Ä¢ Setup webView
2013-01-07 14:30:53.060 Baker[1011:1a003] ‚Ä¢ Loading: book/Page08.html
2013-01-07 14:30:53.086 Baker[1011:1a003] ‚Ä¢ Page did finish load
2013-01-07 14:30:53.111 Baker[1011:1a003] ‚Ä¢ Page did finish load
2013-01-07 14:30:53.112 Baker[1011:1a003] ‚Ä¢ Page did finish load
2013-01-07 14:30:53.199 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:53.199 Baker[1011:1a003]     Page is next page --> load page
2013-01-07 14:30:53.199 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:53.320 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:53.320 Baker[1011:1a003]     Page is next page --> load page
2013-01-07 14:30:53.320 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:53.322 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:53.322 Baker[1011:1a003]     Page is next page --> load page
2013-01-07 14:30:53.322 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:53.324 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:53.324 Baker[1011:1a003]     Page is next page --> load page
2013-01-07 14:30:53.324 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:53.540 Baker[1011:1a003] ‚Ä¢ Page did finish load
2013-01-07 14:30:53.540 Baker[1011:1a003] ‚Ä¢ webView hidden: 0 animating: 1
2013-01-07 14:30:53.541 Baker[1011:1a003] ‚Ä¢ Handle loading of slot -1 with page 15
2013-01-07 14:30:53.541 Baker[1011:1a003] ‚Ä¢ Setup new page for loading
2013-01-07 14:30:53.542 Baker[1011:1a003] ‚Ä¢ Setup webView
2013-01-07 14:30:53.543 Baker[1011:1a003] ‚Ä¢ Loading: book/Page06.html
2013-01-07 14:30:53.576 Baker[1011:1a003]    Handle saved hash reference if necessary
2013-01-07 14:30:53.577 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:53.577 Baker[1011:1a003]     Page is prev page --> load page
2013-01-07 14:30:53.577 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:53.583 Baker[1011:1a003] ‚Ä¢ Page did finish load
2013-01-07 14:30:53.588 Baker[1011:1a003] ‚Ä¢ Page did finish load
2013-01-07 14:30:53.589 Baker[1011:1a003] ‚Ä¢ Page did finish load
2013-01-07 14:30:53.608 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:53.608 Baker[1011:1a003]     Page is prev page --> load page
2013-01-07 14:30:53.608 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:53.610 Baker[1011:1a003] ‚Ä¢ Should webView load the page ?
2013-01-07 14:30:53.610 Baker[1011:1a003]     Page is prev page --> load page
2013-01-07 14:30:53.610 Baker[1011:1a003] ‚Ä¢ Page did start load
2013-01-07 14:30:53.962 Baker[1011:1a003] ‚Ä¢ Page did finish load
2013-01-07 14:30:53.963 Baker[1011:1a003] ‚Ä¢ webView hidden: 0 animating: 1
2013-01-07 14:30:53.987 Baker[1011:1a003] ‚Ä¢ Page did finish load
2013-01-07 14:30:53.987 Baker[1011:1a003] ‚Ä¢ Page did finish load

**Comment by anjimi** [comment]
I'm stumped too, very odd. Our app is about ready to submit to Apple, would be great to fix this before we do, hopefully someone may have an idea...

**Comment by ghost** [comment]
To solve this problem you should do the following:
1. Move notification listeners initialization from viewWillAppear to viewDidLoad method in BakerViewController.m to avoid re-initialization of listeners after exiting the full-screen video playback 
2. Rewrite viewDidAppear method to prevent reloading of current page (See the example below)

``` objc
static BOOL initialAppearance = YES;
- (void)viewDidAppear:(BOOL)animated 
{
    [super viewDidAppear:animated];
    if (!currentPageWillAppearUnderModal) 
    {
        [self willRotateToInterfaceOrientation:self.interfaceOrientation duration:0];
        if(initialAppearance) 
        {
            [self performSelector:@selector(hideBars:) withObject:[NSNumber numberWithBool:YES] afterDelay:0.5];
            [self startReading];
            initialAppearance = NO;
        }
        else 
        {
            [self didRotateFromInterfaceOrientation:self.interfaceOrientation];
        }
    }
    currentPageWillAppearUnderModal = NO;
}
```

PS: sorry for my english :)

**Comment by sempeck** [comment]
@Dmitry-Gruzd thanks for that solution, it works for me.

But there is one little problem. When you quit your magazine and go to shelf, and click to read magazine again - the index is not loaded. After double-tap only the top bar is visible. Would you have any solution to that?

Thanks!

**Comment by folletto** [comment]
That's a different issue, and probably has been already fixed in `master`, see issue #574. ;)

**Comment by sempeck** [comment]
I'm affraid the story here is little bit different.

Here's what's happening:
1. implement Dmitry-Gruzd fix
2. open shelf
3. read a downloaded book (everything works fine - index and full screen video)
4. double tap, click link back to shelf
5. read the same or different book
6. Result: blank page, looks like the first page is not loaded. Pages load after swiping, but there is no load icon nor page title while loading. No index, only navigation bar after double tap.  

I really need that fix to solve my fullscreen video issue, but unfortunatelly it causes different issue. 
I would be grateful for any help in that matter.

**Comment by folletto** [comment]
It's not clear to me... Are you saying that the problem is there already, or appears after you tried to fix it as above?

**Comment by sempeck** [comment]
The problem appears after I do what Dymitry-Gruzd suggested: 
1. Move notification listeners initialization from viewWillAppear to viewDidLoad method in BakerViewController.m to avoid re-initialization of listeners after exiting the full-screen video playback
2. Rewrite viewDidAppear method to prevent reloading of current page

**Comment by folletto** [comment]
Then I can't help much... This issue for us is still open and we didn't check his code. Thanks for testing tho, so when we're able to look into that we already know what are the issues. :)

**Comment by sempeck** [comment]
No worries, it's just the problem with exiting fullscreen that bothers me. I hope you will have a chance to look into that. Thanks again, You guys are doing a great job.

**Comment by ghost** [comment]
Sorry, i made a mistake. In example i wrote static BOOL initialAppearance. In that case you need to set initialAppearance to YES when view did load and unload:

``` objc
static BOOL initialAppearance = YES;
- (void)viewDidLoad {
    initialAppearance = YES;
    ...
}

- (void)viewDidUnload {
    initialAppearance = YES;
    ....
}

- (void)viewDidAppear:(BOOL)animated 
{
    [super viewDidAppear:animated];
    if (!currentPageWillAppearUnderModal) 
    {
        [self willRotateToInterfaceOrientation:self.interfaceOrientation duration:0];
        if(initialAppearance) 
        {
            [self performSelector:@selector(hideBars:) withObject:[NSNumber numberWithBool:YES] afterDelay:0.5];
            [self startReading];
            initialAppearance = NO;
        }
        else 
        {
            [self didRotateFromInterfaceOrientation:self.interfaceOrientation];
        }
    }
    currentPageWillAppearUnderModal = NO;
}
```

Or u can add property (nonatomic, assign) BOOL initialAppearance in BakerViewController and set it to YES only in viewDidLoad method. I hope this will help you

**Comment by benoitchantre** [comment]
I have found that when you close a video that was playing fullscreen, the page scrolls immediately to the top.
Can someone reproduce that?

I have already opened a specific issue for that (#703)

**Comment by Simbul** [comment]
This specific issue should now be closed. Please @anjimi, @Mediorama, @Dmitry-Gruzd, can you double check to see if this fixes it for you?
(Note that you'll still have to manually remove the user agent)

**Comment by anjimi** [comment]
I've tested this with the latest master version and can confirm it is fixed which is fantastic - we can now look to release our app for iphone. Thank you everyone for your work on this. :)

However there is a new/lesser bug in iOS 5.0/5.1 only:

When exiting fullscreen video the app screen moves up by 20px (the same effect as was happening in https://github.com/Simbul/baker/issues/584 which is now fixed). Oddly, if the device has been rotated before playing fullscreen video it doesn't happen.

It works fine in iOS 6.1.
